modules/swift/examples/code_snippets/SampleCodeTest.swift

```diff
@@ -1209,2824 +1208,1603 @@ class SampleCodeTest {
-        if !self.keyChainAccessAllowed {
-            return
-        }
-
-// tag::xctListener-auth-tls-CCA-Root-full[]
-// tag::xctListener-auth-tls-CCA-Root[]
+        // tag::xctListener-auth-tls-CCA-Root-full[]
+        // tag::xctListener-auth-tls-CCA-Root[]
         // Root Cert:
-        let rootCertData = try dataFromResource(name: "identity/client-ca", ofType: "der")
+        let path = Bundle.main.path(forResource: "identity/client-ca", ofType: "der")
+        let rootCertData = try NSData(contentsOfFile: path!, options: []) as Data
         let rootCert = SecCertificateCreateWithData(kCFAllocatorDefault, rootCertData as CFData)!
-
+        
         // Listener:
-        let listenerAuth = ListenerCertificateAuthenticator.init(rootCerts: [rootCert])
-        let listener = try listen(tls: true, auth: listenerAuth)
-// end::xctListener-auth-tls-CCA-Root[]
-
-        // Cleanup:
-        try TLSIdentity.deleteIdentity(withLabel: clientCertLabel)
-
+        var config = URLEndpointListenerConfiguration(database: otherDB)
+        config.authenticator = ListenerCertificateAuthenticator.init(rootCerts: [rootCert])
+        self.listener = URLEndpointListener(config: config)
+        
+        try self.listener.start()
+        // end::xctListener-auth-tls-CCA-Root[]
+        
+        
         // Create client identity:
-        let clientCertData = try dataFromResource(name: "identity/client", ofType: "p12")
-        let identity = try TLSIdentity.importIdentity(withData: clientCertData, password: "123", label: clientCertLabel)
-
+        let clientCertPath = Bundle.main.path(forResource: "identity/client", ofType: "p12")
+        let clientCertData = try NSData(contentsOfFile: clientCertPath!, options: []) as Data
+        let identity = try TLSIdentity.importIdentity(withData: clientCertData,
+                                                      password: "123",
+                                                      label: "couchbaselite-client-cert-label")
+        
         // Replicator:
-        let auth = ClientCertificateAuthenticator.init(identity: identity)
-        let serverCert = listener.tlsIdentity!.certs[0]
-
-        self.ignoreException {
-            self.run(target: listener.localURLEndpoint, type: .pushAndPull, continuous: false, auth: auth, serverCert: serverCert)
-// end::xctListener-auth-tls-CCA-Root-full[]
-        }
-
-        // Cleanup:
-        try TLSIdentity.deleteIdentity(withLabel: clientCertLabel)
-        try stopListen()
+        let target = URLEndpoint(url: URL(string: "wss://localhost:4985/otherDB")!)
+        var replicatorConfig = ReplicatorConfiguration(database: database, target: target)
+        replicatorConfig.replicatorType = .pushAndPull
+        replicatorConfig.continuous = false
+        replicatorConfig.authenticator = ClientCertificateAuthenticator.init(identity: identity)
+        replicatorConfig.pinnedServerCertificate = self.listener.tlsIdentity!.certs[0]
```

Comment:
This doesn't work for real life. The listener and replicator are not in the same code.

Reply:
Please add a comment that in the real application, the cert needs to send over from the listener app to the client to app to pin.

Reply:
As per Ian, these snippets are not used! I am removing all `xctListener` tags.